torch/csrc/jit/tensorexpr/kernel.cpp

```diff
@@ -963,6 +963,17 @@ std::vector<torch::jit::StrideInput>& TensorExprKernel::
   TORCH_INTERNAL_ASSERT(false);
 }
 
+torch::jit::StrideInput TensorExprKernel::getSymbolicOutputStrideDesc(
+    const torch::jit::Value* value) {
+  for (size_t i : c10::irange(graph_->outputs().size())) {
+    if (value == graph_->outputs().at(i)) {
+      TORCH_INTERNAL_ASSERT(sym_stride_outputs_.count(i));
+      return sym_stride_outputs_[i];
+    }
+  }
+  TORCH_INTERNAL_ASSERT(false);
+}
```

Comment:
Hm, this is getting hairy, I wonder if we can make this more elegant (ie. remove duplication of how we treat inputs and outputs and avoid searching through the vector each time to just locate Value). I think this issue is beyond the scope of this PR, but if you have ideas of how it could be improved - that would be awesome :)

Reply:
I added `symbolic_strides` to `TensorExprKernel` as an attribute. The `symbolic_strides` contains the symbolic stride information of the inputs and outputs. https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/tensorexpr/kernel.cpp#L1653 So we don't need to handle inputs and output separately. But I'm not sure if I should refine more code in this PR based on this change, as I plan to make this PR more focused on channels-last and submit another PR to handle input and output in unison.

Reply:
Yeah, it can be in another PR!